packages/period-selector-dialog/src/modules/__tests__/FixedPeriodsGenerator.spec.js

```diff
@@ -0,0 +1,507 @@
+                startDate: '2018-12-31',
+                endDate: '2019-01-13',
+                name: 'Bi-Week 1 - 2018-12-31 - 2019-01-13',
+                iso: '2019BiW1',
+                id: '2019BiW1',
+            });
+        });
+
+        it('should return the correct object for bi-week 27', () => {
+            expect(periods[26]).toEqual({
+                startDate: '2019-12-30',
+                endDate: '2020-01-12',
+                name: 'Bi-Week 27 - 2019-12-30 - 2020-01-12',
+                iso: '2019BiW27',
+                id: '2019BiW27',
+            });
+        });
+    });
+
+    describe('Monthly period generator', () => {
+        let periods;
+
+        beforeAll(() => {
+            const generator = periodsGenerator.get('Monthly');
+
+            periods = generator.generatePeriods({
+                offset: 2019 - (new Date()).getFullYear(),
+                filterFuturePeriods: false,
+                reversePeriods: false,
+            });
+        });
+
+        it('should return the correct number of months in 2019', () => {
+                expect(periods.length).toEqual(12);
+        });
+
+        it('should return the correct object for month 1', () => {
+            expect(periods[0]).toEqual({
+                startDate: '2019-01-01',
+                endDate: '2019-01-31',
+                name: 'January 2019',
+                iso: '201901',
+                id: '201901',
+            });
+        });
+
+        it('should return the correct object for bi-week 26', () => {
+            expect(periods[11]).toEqual({
+                startDate: '2019-12-01',
+                endDate: '2019-12-31',
+                name: 'December 2019',
+                iso: '201912',
+                id: '201912',
+            });
+        });
+    });
+
+    describe('Bi-Monthly period generator', () => {
+        let periods;
+
+        beforeAll(() => {
+            const generator = periodsGenerator.get('Bi-monthly');
+
+            periods = generator.generatePeriods({
+                offset: 2019 - (new Date()).getFullYear(),
+                filterFuturePeriods: false,
+                reversePeriods: false,
+            });
+        });
+
+        it('should return the correct number of bi-months in 2019', () => {
+                expect(periods.length).toEqual(6);
+        });
+
+        it('should return the correct object for bi-month 1', () => {
+            expect(periods[0]).toEqual({
+                startDate: '2019-01-01',
+                endDate: '2019-02-28',
+                name: 'January - February 2019',
+                iso: '201901B',
+                id: '201901B',
+            });
+        });
+
+        it('should return the correct object for bi-month 6', () => {
+            expect(periods[5]).toEqual({
+                startDate: '2019-11-01',
+                endDate: '2019-12-31',
+                name: 'November - December 2019',
+                iso: '201911B',
```

Comment:
Are the iso and id values correct?  I would have thought '201906B'

Reply:
They are correct, cfr documentation and api:
https://docs.dhis2.org/master/en/developer/html/webapi_date_perid_format.html#
https://play.dhis2.org/dev/api/32/periodTypes.json

Reply:
It was indeed wrong. Fixed.